corehq/apps/app_manager/ui_translations/ui_translations.py

```diff
@@ -46,16 +46,41 @@ def process_ui_translation_upload(app, trans_file):
     default_trans = get_default_translations_for_download(app, commcare_version)
     lang_with_defaults = app.langs[get_index_for_defaults(app.langs)]
 
+    # Find all of the ${0}-looking values in a translation.
+    # Returns list like ["${0}", "${1}", ...]
+    def _get_params(translation):
+        if not translation:
+            return []
+        return re.findall(r"\$.*?}", translation)
+
+    # Create a string version of all parameters in a translation, suitable both for
+    # comparison and for displaying in an error message.
+    def _get_params_text(params):
+        if not params:
+            return "no parameters"
+        return ", ".join(["${{{}}}".format(num) for num in sorted({re.sub(r'\D', '', p) for p in params})])
```

Comment:
this is quite a complex operation in iteration, can i push you to open this up in multiline for loop, with comments(if needed)?
will sorting here result in change of order of things?

Reply:
Updated in 74dfabfe1da

The sorting is only for comparing the new and old translations, it doesn't affect the actual translation.